include/sys/util: Fix WAIT_FOR for POSIX_ARCH

aescolar opened the pull request:
WAIT_FOR is a busy wait loop, which in the POSIX ARCH should include a very minor delay in each iteration.

After this fix, tests/lib/sys_util does not need to exclude native_posix anymore.

andyross commented on include/zephyr/sys/util.h:
```diff
@@ -578,6 +578,15 @@ char *utf8_lcpy(char *dst, const char *src, size_t n);
+ */
+#else
+#define __z_POSIX_ARCH_WAIT_A_BIT
+#endif
```
If this is an API in a general header, can we name it something generic?  "_Z_SPIN_DELAY" or something, with a comment explaining the requirement on POSIX that it be present in any spin loop?

aescolar replied:
I did not mean it as an API, but as a hidden macro for WAIT_FOR below. We can add such a general purpose macro as you mention though.

aescolar replied:
Ok, any preferred name?

stephanosio replied:
Since we are talking about naming here, please do not use underscores since they are "reserved."

aescolar replied:
Z_SPIN_DELAY()? Otherwise just give a name  :)

andyross replied:
Your code, your call.  I think Z_SPIN_DELAY is great personally, but I'm not picky (even about underscores, heh).

aescolar committed:
util: Provide macro to cause a minor delay in busy loops in POSIX ARCH

In the POSIX ARCH, code takes zero time to execute,
so busy wait loops become infinite loops, unless we
force the loop to take a bit of time.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
include/sys/util: Fix WAIT_FOR for POSIX_ARCH

WAIT_FOR is a busy wait loop, which in the POSIX ARCH
should include a very minor delay in each iteration.

After this fix, tests/lib/sys_util does not need to
exclude native_posix anymore.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>
